python/xorbits/numpy/numpy_adapters/tests/test_numpy_adapters.py

```diff
@@ -19,60 +19,126 @@
     ],
 )
-def test_zero_param_funcs(mod_name, func):
+def test_numpy_fallback(mod_name, func, params):
     with pytest.warns(Warning) as w:
-        xnp_func = getattr(getattr(xnp, mod_name), func)
-        np_func = getattr(getattr(np, mod_name), func)
-
-        res = xnp_func()
-        assert f"xorbits.numpy.{mod_name}.{func} will fallback to NumPy" == str(
-            w[0].message
+        xnp_func = (
+            getattr(getattr(xnp, mod_name), func)
+            if mod_name != ""
+            else getattr(xnp, func)
+        )
+        np_func = (
+            getattr(getattr(np, mod_name), func)
+            if mod_name != ""
+            else getattr(np, func)
         )
 
-        xnp_output = res.execute().fetch()
-        np_output = np_func()
+        xnp_output = xnp_func(*params).execute().fetch()
+        np_output = np_func(*params)
 
+        assert (
+            f"xorbits.numpy{'.' + mod_name if mod_name != '' else ''}.{func} will fallback to NumPy"
+            == str(w[0].message)
+        )
+
+        if isinstance(xnp_output, int):
+            xnp_output = np.int64(xnp_output)
+        if isinstance(xnp_output, float):
+            xnp_output = np.float64(xnp_output)
         assert type(xnp_output) == type(np_output)
 
         if isinstance(np_output, np.ndarray):
             assert isinstance(xnp_output, np.ndarray)
-            assert np.equal(xnp_output, np_output)
+            assert np.equal(xnp_output.all(), np_output.all())
         if isinstance(np_output, object):
             assert isinstance(xnp_output, object)
             assert dir(xnp_output) == dir(np_output)
 
 
-@pytest.mark.parametrize(
-    "mod_name,func",
-    [
-        ("random", "Generator"),
-    ],
-)
-def test_one_param_funcs(mod_name, func):
+def test_tensorsolve_fallback(setup):
     with pytest.warns(Warning) as w:
-        xnp_func = getattr(getattr(xnp, mod_name), func)
-        np_func = getattr(getattr(np, mod_name), func)
+        a = np.eye(2 * 3 * 4)
+        a.shape = (2 * 3, 4, 2, 3, 4)
+        b = np.random.randn(2 * 3, 4)
 
-        res = xnp_func(np.random.PCG64())
-        assert f"xorbits.numpy.{mod_name}.{func} will fallback to NumPy" == str(
+        xnp_output = xnp.linalg.tensorsolve(a, b).execute().fetch()
+        np_output = np.linalg.tensorsolve(a, b)
+
+        assert f"xorbits.numpy.linalg.tensorsolve will fallback to NumPy" == str(
             w[0].message
         )
 
-        xnp_output = res.execute().fetch()
-        np_output = np_func(np.random.PCG64())
+        assert np.equal(xnp_output.all(), np_output.all())
 
-        assert type(xnp_output) == type(np_output)
 
-        if isinstance(np_output, np.ndarray):
-            assert isinstance(xnp_output, np.ndarray)
-            assert np.equal(xnp_output, np_output)
-        if isinstance(np_output, object):
-            assert isinstance(xnp_output, object)
-            assert dir(xnp_output) == dir(np_output)
+def test_tensorinv_fallback(setup):
```

Comment:
Better have a test for docstring. You may refer to the pandas adapter docstring test case.